### PR TITLE
Make Mountain Car example deterministic

### LINKED ISSUE #202: Mountain car example is not deterministic
Running the mountain_car example twice in a row yields different results:
mountain_car

In addition, I also observed runs where the environment was not solved by the found champion.

I think the example should behave deterministic, so we should fix this.

### PR DESCRIPTION
**Pull request by jakobj**
building on #204 this PR makes the Mountain Car example reproducible by fixing the environment seed before the first observation is obtained. in addition, it fixes a couple of minor issues in this example and slightly adapts parameters to make sure a good solution is found in reasonable time.

closes #202

### CODE REVIEW

File: examples/example_mountain_car.py [hunk 1]
```diff
@@ -55,30 +51,30 @@ def __init__(self, idx, inputs):
 # episodes.
 
 
-def inner_objective(f, seed, n_total_steps, *, render):
-
-    np.random.seed(seed)
+def inner_objective(f, seed, n_trials_per_individual, n_total_steps, *, render):
```
Comment (HenrikMettler) [inline examples/example_mountain_car.py:54]: Wouldn't `n_episodes_per_individual` be proper RL-lingo?

Comment (HenrikMettler) [reply examples/example_mountain_car.py:54]: ah I saw below that fast agents have more episodes, if they don't use all time steps. But I think it is a bit confusing that trial is bigger than episode. Maybe `n_runs_per_individual`

File: examples/example_mountain_car.py [hunk 2]
```diff
@@ -106,12 +102,14 @@ def objective(ind, seed, n_total_steps):
...
-            cum_reward_all_episodes = inner_objective(f, seed, n_total_steps, render=False)
+            cum_reward_all_episodes = inner_objective(
+                f, seed, n_trials_per_individual, n_total_steps, render=False
+            )
 
         # more episodes are better, more reward is better
         n_episodes = float(len(cum_reward_all_episodes))
         mean_cum_reward = np.mean(cum_reward_all_episodes)
-        ind.fitness = n_episodes + 1.0 * mean_cum_reward
+        ind.fitness = n_episodes / n_trials_per_individual + mean_cum_reward
```
Comment (HenrikMettler) [inline examples/example_mountain_car.py:112]: The first summand is a bit confusing, but renaming the variable as suggested above is enough I guess

File: examples/example_mountain_car.py (hunk 1 above)
Comment (jakobj) [reply examples/example_mountain_car.py:54]: yeah, this might indeed be a bit confusing naming. i'll use your suggestion :+1:

**Comment by jakobj** [comment]
thanks! squashed and waiting for travis :)

**Pull request merged**